pkg/scale/decode.go

```diff
@@ -60,26 +59,23 @@ func indirect(dstv reflect.Value) (elem reflect.Value) {
 func Unmarshal(data []byte, dst interface{}) (err error) {
 	dstv := reflect.ValueOf(dst)
 	if dstv.Kind() != reflect.Ptr || dstv.IsNil() {
-		err = fmt.Errorf("unsupported dst: %T, must be a pointer to a destination", dst)
+		err = fmt.Errorf("%w: %T", ErrUnsupportedDST, dst)
 		return
 	}
 
 	elem := indirect(dstv)
-	if err != nil {
-		return
-	}
 
 	buf := &bytes.Buffer{}
 	ds := decodeState{}
 	_, err = buf.Write(data)
 	if err != nil {
-		return
+		return fmt.Errorf("writing data: %w", err)
 	}
 	ds.Reader = buf
 
 	err = ds.unmarshal(elem)
 	if err != nil {
-		return
+		return fmt.Errorf("unmarshaling %v: %w", elem, err)
```

Comment:
meganit I think it's `unmarshalling` in UK english?? I feel I remember the linter shouting at me for that one